funding/manager.go

```diff
@@ -653,6 +668,116 @@ func (m *Manager) RemovePendingBatchArtifacts(
 	return nil
 }
 
+// OfferSidecar creates a sidecar channel offer and embeds it in a new sidecar
+// ticket. The offer is signed with the local lnd's node public key.
+func (m *Manager) OfferSidecar(ctx context.Context, capacity,
+	pushAmt btcutil.Amount, nodePubKey [33]byte) (*sidecar.Ticket, error) {
+
+	if capacity == 0 || capacity%order.BaseSupplyUnit != 0 {
+		return nil, fmt.Errorf("channel capacity must be positive "+
+			"multiple of %d", order.BaseSupplyUnit)
+	}
+
+	if pushAmt >= capacity {
+		return nil, fmt.Errorf("push amount must be smaller than " +
+			"capacity")
+	}
+
+	// Because of the channel reserves, the push amount cannot be greater
+	// than capacity-commit_fee. Because we can't predict the commit fees we
+	// cap the push amount to 50% of the capacity here as a safety measure.
+	if pushAmt > (capacity)/2 {
+		return nil, fmt.Errorf("push amount cannot exceed 50%% of " +
+			"channel capacity")
+	}
+
+	// We'll be signing the offer with the local lnd node's public key.
+	nodeKey, err := btcec.ParsePubKey(nodePubKey[:], btcec.S256())
+	if err != nil {
+		return nil, fmt.Errorf("error parsing node pubkey: %v", err)
+	}
+
+	// So far everything looks good. Let's create the ticket with the offer
+	// now.
+	ticket, err := sidecar.NewTicket(
+		sidecar.VersionDefault, capacity, pushAmt, nodeKey,
+	)
+	if err != nil {
+		return nil, fmt.Errorf("error creating sidecar ticket: %v", err)
+	}
+
+	// Let's sign the offer part of the ticket with our node's identity key
+	// now.
+	digest, err := ticket.OfferDigest()
+	if err != nil {
+		return nil, fmt.Errorf("error digesting offer: %v", err)
+	}
+	rawSig, err := m.SignerClient.SignMessage(
+		ctx, digest[:], nodeIdentityKeyLoc,
+	)
+	if err != nil {
+		return nil, fmt.Errorf("error signing offer: %v", err)
+	}
+	wireSig, err := lnwire.NewSigFromRawSignature(rawSig)
+	if err != nil {
+		return nil, fmt.Errorf("error parsing raw signature: %v", err)
+	}
+	ecSig, err := wireSig.ToSignature()
+	if err != nil {
+		return nil, fmt.Errorf("error parsing EC signature: %v", err)
+	}
+
+	// Let's now store and return the ticket with the signed offer.
+	ticket.Offer.SigOfferDigest = ecSig
+	err = m.DB.AddSidecar(ticket)
+	if err != nil {
+		return nil, fmt.Errorf("error storing sidecar ticket: %v", err)
+	}
+
+	return ticket, nil
+}
+
+// EncodeOrderedTicket signs the order digest of a sidecar ticket and returns it
+// in the string encoded version.
+func (m *Manager) EncodeOrderedTicket(ctx context.Context, t *sidecar.Ticket,
+	nonce order.Nonce) (string, error) {
```

Comment:
Isn't the nonce already encoded in the ticket itself?

Reply:
Also this is the _bid_ nonce right?

Reply:
Ended up solving this differently.